homeassistant/components/nintendo_parental/config_flow.py

```diff
@@ -0,0 +1,58 @@
+"""Config flow for the Nintendo Switch Parental Controls integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING, Any
+
+from pynintendoparental import Authenticator
+from pynintendoparental.exceptions import HttpException, InvalidSessionTokenException
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_API_TOKEN
+
+from .const import CONF_SESSION_TOKEN, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class NintendoConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Nintendo Switch Parental Controls."""
+
+    def __init__(self) -> None:
+        """Initialize a new config flow instance."""
+        self.auth = Authenticator.generate_login()
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            try:
+                await self.auth.complete_login(
+                    self.auth, user_input[CONF_API_TOKEN], False
+                )
+            except (ValueError, InvalidSessionTokenException, HttpException):
+                errors["base"] = "invalid_auth"
+            except Exception as exc:
+                _LOGGER.exception("Unknown error during setup", exc_info=exc)
```

Comment:
```suggestion
                _LOGGER.exception("Unknown error during setup")
```
I believe this already happens